.github/workflows/link-check-workflow.yml

```diff
@@ -0,0 +1,22 @@
+name: Link Checker
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  linkchecker:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: lychee Link Checker
+        id: lychee
+        uses: lycheeverse/lychee-action@master
+        with:
+          args: --accept=200,403,429 --exclude=localhost "**/*.html" "**/*.md" "**/*.txt" "**/*.json"
```

Comment:
From https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403
> The HTTP 403 Forbidden client error status response code indicates that the server understood the request but refuses to authorize it.

Should we consider 403 as acceptable?

Reply:
This is simply based off of the recommendation from [here](https://github.com/opensearch-project/opensearch-plugins/pull/54). I think the purpose is more for checking purely broken links, rather than permissions-issues-related one, which could possibly be related to the way the github action is pinging/visiting the existing links.